BuildTasks/Common/Common.ts

```diff
@@ -334,17 +336,44 @@ function getTaskPathContributions(manifest: any): string[] {
         .map((c: any) => c.properties["name"]);
 }
 
-function updateTaskId(manifest: any, publisherId: string, extensionId: string): Object {
+function updateTaskId(manifest: any, publisherId: string, extensionId: string): unknown {
     tl.debug(`Task manifest ${manifest.name} id before: ${manifest.id}`);
 
-    let extensionNs = uuidv5("url", "https://marketplace.visualstudio.com/vsts", true);
+    const extensionNs = uuidv5("url", "https://marketplace.visualstudio.com/vsts", true);
     manifest.id = uuidv5(extensionNs, `${publisherId}.${extensionId}.${manifest.name}`, false);
 
     tl.debug(`Task manifest ${manifest.name} id after: ${manifest.id}`);
     return manifest;
 }
 
-function updateTaskVersion(manifest: any, extensionVersionString: string, extensionVersionType: string): Object {
+function updateExtensionManifestTaskIds(manifest: any, originalTaskId: string, newTaskId: string): unknown {
+    if (!manifest.contributions) {
+        tl.debug(`No contributions found`);
+        return manifest;
+    }
+    
+    manifest.contributions
+        .filter((c: any) => c.type !== "ms.vss-distributed-task.task" && c.properties && c.properties.supportsTasks)
```

Comment:
I initially had the type filter on the task type as well but the multiping demo actually has the supportsTasks in a task contribution for some reason. I have no idea if that even works, but I removed that check before because otherwise the multiping demo is not supported.